src/contracts/libraries/SlashingLib.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+/// @dev the stakerScalingFactor and operatorMagnitude have initial default values to 1e18 as "1"
+/// to preserve precision with uint256 math. We use `WAD` where these variables are used
+/// and divide to represent as 1
+uint64 constant WAD = 1e18;
+
+/*
+ * There are 2 types of shares:
+ *      1. depositShares
+ *          - These can be converted to an amount of tokens given a strategy
+ *              - by calling `sharesToUnderlying` on the strategy address (they're already tokens 
+ *              in the case of EigenPods)
+ *          - These live in the storage of EPM and SM strategies 
+ *      2. shares
+ *          - For a staker, this is the amount of shares that they can withdraw
+ *          - For an operator, this is the sum of its staker's withdrawable shares       
+ * 
+ * Note that `withdrawal.scaledSharesToWithdraw` is scaled for the beaconChainETHStrategy to divide by the beaconChainScalingFactor upon queueing
+ * and multiply by the beaconChainScalingFactor upon withdrawal
+ */
+
+struct StakerScalingFactors {
+    uint256 depositScalingFactor;
+    // we need to know if the beaconChainScalingFactor is set because it can be set to 0 through 100% slashing
+    bool isBeaconChainScalingFactorSet;
+    uint64 beaconChainScalingFactor;
+}
+
+using SlashingLib for StakerScalingFactors global;
+
+// TODO: validate order of operations everywhere
+library SlashingLib {
+    using Math for uint256;
+    using SlashingLib for uint256;
+
+    // WAD MATH
+
+    function mulWad(uint256 x, uint256 y) internal pure returns (uint256) {
+        return x.mulDiv(y, WAD);
+    }
+
+    function divWad(uint256 x, uint256 y) internal pure returns (uint256) {
+        return x.mulDiv(WAD, y);
+    }
+
+    // GETTERS
+
+    function getDepositScalingFactor(
+        StakerScalingFactors memory ssf
+    ) internal pure returns (uint256) {
+        return ssf.depositScalingFactor == 0 ? WAD : ssf.depositScalingFactor;
+    }
+
+    function getBeaconChainScalingFactor(
+        StakerScalingFactors memory ssf
+    ) internal pure returns (uint64) {
+        return
+            !ssf.isBeaconChainScalingFactorSet && ssf.beaconChainScalingFactor == 0 ? WAD : ssf.beaconChainScalingFactor;
```

Comment:
Can this be simplified to:

```
return ssf.isBeaconChainScalingFactorSet ? ssf.beaconChainScalingFactor : WAD;
```

Reply:
Followup: the code seems to imply that it's possible for `beaconChainScalingFactor` to equal 0. However, this would cause `mulDiv` to revert if it's used as the denominator!

If this _can_ be zero, we need to fix that.

If this _can't_ be zero, this code is incorrect.